cypress/integration/sales/void_a_sales_invoice_then_invoice_the_billing_candidates_again.js

```diff
@@ -0,0 +1,372 @@
+      ShipmentDispositions.getRows().should('have.length', 1);
+
+      // select the first Table Row and click it (open it)
+      // eslint-disable-next-line prettier/prettier
+      ShipmentDispositions
+        .getRows()
+        .eq(0)
+        .dblclick();
+    });
+
+    it('Shipment Disposition checks', function() {
+      cy.getStringFieldValue('C_BPartner_ID').should('contain', businessPartnerName);
+      cy.getStringFieldValue('M_Product_ID').should('contain', productName);
+      cy.getStringFieldValue('C_Order_ID').should('equal', salesOrderDocumentNumber);
+      cy.getStringFieldValue('QtyOrdered_Calculated').should('equal', originalQuantity.toString(10));
+      cy.getStringFieldValue('QtyToDeliver ').should('equal', originalQuantity.toString(10));
+    });
+
+    it('Execute action "Generate shipments', function() {
+      cy.executeHeaderActionWithDialog('M_ShipmentSchedule_EnqueueSelection');
+      cy.selectInListField('QuantityType', 'Quantity to deliver', true, null, true);
+      cy.setCheckBoxValue('IsCompleteShipments', true, true);
+      cy.setCheckBoxValue('IsShipToday', false, true);
+      cy.pressStartButton();
+      cy.getNotificationModal(shipmentNotificationModalText);
+      cy.waitUntilProcessIsFinished();
+      cy.getDOMNotificationsNumber().should('equal', 1);
+      // todo check notification inbox text!
+      cy.readAllNotifications();
+    });
+
+    it('Open the Referenced Billing Candidates', function() {
+      salesOrders.visit(salesOrderRecordId);
+      cy.openReferencedDocuments('C_Order_C_Invoice_Candidate');
+    });
+
+    it('Expect only 1 Billing Candidates row and open it', function() {
+      BillingCandidates.getRows().should('have.length', 1);
+
+      // select the first Table Row and click it (open it)
+      // eslint-disable-next-line prettier/prettier
+      BillingCandidates
+        .getRows()
+        .eq(0)
+        .dblclick();
+    });
+
+    it('Billing Candidates checks after Shipment completion', function() {
+      const qtyDelivered = originalQuantity.toString(10);
+      const qtyInvoiced = '0';
+
+      checkBillingCandidate(qtyDelivered, qtyInvoiced);
+    });
+  });
+
+  describe('Generate Invoice and check it', function() {
+    it('Execute action "Generate Invoices"', function() {
+      cy.executeHeaderActionWithDialog('C_Invoice_Candidate_EnqueueSelectionForInvoicing');
+      cy.pressStartButton(500);
+
+      cy.getNotificationModal(generateInvoicesNotificationModalText);
+      cy.waitUntilProcessIsFinished();
+      cy.getDOMNotificationsNumber().should('equal', 1);
+      // todo check notification inbox text!
+      // expected text:
+      // Rechnung 145808 für Partner G0002 Test Lieferant 1 wurde erstellt.
+      cy.readAllNotifications();
+    });
+
+    it('Billing Candidates checks after Invoice generation', function() {
+      const qtyDelivered = originalQuantity.toString(10);
+      const qtyInvoiced = originalQuantity.toString(10);
```

Comment:
Jesus I hate the fact that we have to use strings. btw I didn't even know `toString` accepts a radix parameter ;)

Reply:
I'm using toString because i couldn't find a way to `equals` with `==` instead of `===` as cypress is doing now.

Reply:
I meant `.should('equals', qtyDelivered)` the `equals` expects a string in there (and i assume it uses `===` for the equals`) and not a number (if a number were accepted, then `==` would be ok as `123.456` == `'123.456'`, as far as i know)